pkg/user/tx_client.go

```diff
@@ -410,8 +411,10 @@ func (client *TxClient) BroadcastTx(ctx context.Context, msgs []sdktypes.Msg, op
 	return client.broadcastTxAndIncrementSequence(ctx, client.conns[0], txBytes, account)
 }
 
+// broadcastTxAndIncrementSequence submits a transaction to the chain and increments the sequence of the signer.
+// It will retry the transaction if it encounters a sequence mismatch.
 func (client *TxClient) broadcastTxAndIncrementSequence(ctx context.Context, conn *grpc.ClientConn, txBytes []byte, signer string) (*sdktypes.TxResponse, error) {
-	resp, err := client.broadcastTx(ctx, conn, txBytes, signer)
+	resp, err := client.broadcastTxWithRetry(ctx, conn, txBytes, signer)
```

Comment:
so we're potentially hitting an incorrect sequence here, but then increasing?

what happens if the retry also fails?

Reply:
is there any chance or place that we will continually retry indefinitely?